elastic/cloudbeat
# Eks basic integration tests for cloudbeat

Added support for executing cloudbeat integration tests on EKS.
Tests execution - **Production Mode** (Pytest docker is running inside EKS cluster)
1. Provision EKS cluster using aws or eksctl tools
2. Select EKS context `kubectl config use-context <your_eks_context>`
3. Deploy tests environment: `just deploy-local-tests-helm pre_merge`
4. Apply AWS certificates: `kubectl create secret generic aws-creds-secret -n kube-system --from-literal=AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY_ID --from-literal=AWS_SECRET_ACCESS_KEY=$AWS_SECRET_ACCESS_KEY `
5. Execute tests `just run-tests`
6. If all tests passed test-pod status will be succeeded, otherwise read logs from test-pod.
 
Tests execution - **Tests Dev Mode** (Tests execution is done on local machine)
1. Repeat step 1-4 from **Production mode**
2. Perform port forward for ES: `kubectl port-forward svc/elasticsearch-master -n kube-system 9200`
3. Execute tests `poetry run pytest -s -v -m "pre_merge" --alluredir=./allure/results/ --clean-alluredir`

If changing cloudbeat code:
- Build Cloudbeat image and upload to ECR, note that for chip M1 specify GOARCH=amd64 and for docker -> --platform linux/amd64

If changing tests code:
- Build pytest docker `just build-pytest-docker`
- Load pytest docker to ECR `just load-pytest-eks`
